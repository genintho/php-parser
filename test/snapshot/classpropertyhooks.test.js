const parser = require("../main");
//
describe("classpropertyhooks", () => {
  it("not supported in php < 8.4", () => {
    expect(() => {
      parser.parseEval(
        `
        class BookViewModel
        {
            public string $credits {
              get => 'mailto:' . $this->email;
            }
        }
        `,
        {
          parser: {
            version: "8.3",
          },
        },
      );
    }).toThrowErrorMatchingSnapshot();
  });

  it("getter arrow function", () => {
    expect(
      parser.parseEval(
        `
        class BookViewModel
        {
            public string $credits {
              get => 'mailto:' . $this->email;
            }
        }
        `,
        {
          parser: {
            version: "8.4",
          },
        },
      ),
    ).toMatchSnapshot();
  });

  it("getter block", () => {
    expect(
      parser.parseEval(
        `
        class BookViewModel
        {
          public string $credits {
            get {
              'mailto:' . $this->email;
            }
          }
        }
        `,
        {
          parser: {
            version: "8.4",
          },
        },
      ),
    ).toMatchSnapshot();
  });
});
