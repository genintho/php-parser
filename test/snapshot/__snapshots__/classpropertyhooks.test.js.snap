// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classpropertyhooks getter arrow function 1`] = `
Program {
  "children": [
    Class {
      "attrGroups": [],
      "body": [
        PropertyStatement {
          "isStatic": false,
          "kind": "propertystatement",
          "properties": [
            Property {
              "attrGroups": [],
              "hooks": [
                PropertyHook {
                  "body": Bin {
                    "kind": "bin",
                    "left": String {
                      "isDoubleQuote": false,
                      "kind": "string",
                      "raw": "'mailto:'",
                      "unicode": false,
                      "value": "mailto:",
                    },
                    "right": PropertyLookup {
                      "kind": "propertylookup",
                      "offset": Identifier {
                        "kind": "identifier",
                        "name": "email",
                      },
                      "what": Variable {
                        "curly": false,
                        "kind": "variable",
                        "name": "this",
                      },
                    },
                    "type": ".",
                  },
                  "kind": "propertyhooks",
                  "name": "get",
                },
              ],
              "kind": "property",
              "name": Identifier {
                "kind": "identifier",
                "name": "credits",
              },
              "nullable": false,
              "readonly": false,
              "type": TypeReference {
                "kind": "typereference",
                "name": "string",
                "raw": "string",
              },
              "value": null,
            },
          ],
          "visibility": "public",
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "isReadonly": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "BookViewModel",
      },
    },
  ],
  "errors": [],
  "kind": "program",
}
`;

exports[`classpropertyhooks getter block 1`] = `
Program {
  "children": [
    Class {
      "attrGroups": [],
      "body": [
        PropertyStatement {
          "isStatic": false,
          "kind": "propertystatement",
          "properties": [
            Property {
              "attrGroups": [],
              "hooks": [
                PropertyHook {
                  "body": Block {
                    "children": [
                      ExpressionStatement {
                        "expression": Bin {
                          "kind": "bin",
                          "left": String {
                            "isDoubleQuote": false,
                            "kind": "string",
                            "raw": "'mailto:'",
                            "unicode": false,
                            "value": "mailto:",
                          },
                          "right": PropertyLookup {
                            "kind": "propertylookup",
                            "offset": Identifier {
                              "kind": "identifier",
                              "name": "email",
                            },
                            "what": Variable {
                              "curly": false,
                              "kind": "variable",
                              "name": "this",
                            },
                          },
                          "type": ".",
                        },
                        "kind": "expressionstatement",
                      },
                    ],
                    "kind": "block",
                  },
                  "kind": "propertyhooks",
                  "name": "get",
                },
              ],
              "kind": "property",
              "name": Identifier {
                "kind": "identifier",
                "name": "credits",
              },
              "nullable": false,
              "readonly": false,
              "type": TypeReference {
                "kind": "typereference",
                "name": "string",
                "raw": "string",
              },
              "value": null,
            },
          ],
          "visibility": "public",
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "isReadonly": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "BookViewModel",
      },
    },
  ],
  "errors": [],
  "kind": "program",
}
`;

exports[`classpropertyhooks not supported in php < 8.4 1`] = `"Parse Error: Typed Class Constants requires PHP 8.4+ on line 5"`;
